lib/Cake/View/Helper/FlashHelper.php

```diff
@@ -82,10 +82,15 @@ public function render($key = 'flash', $options = array()) {
 			));
 		}
 
-		$flash = $options + $flash;
 		CakeSession::delete("Message.$key");
-		$flash['key'] = $key;
 
-		return $this->_View->element($flash['element'], $flash);
+		$out = '';
+		foreach ($flash as $message) {
+			$message['key'] = $key;
+			$message = $options + $message;
+			$out .= $this->_View->element($message['element'], $message);
```

Comment:
Nitpicking: Could we add a new line after/before each new element? Would produce nice HTML.